dali/operators/util/randomizer.cuh

```diff
@@ -18,52 +18,51 @@
 
 #include <math.h>
 #include "dali/core/device_guard.h"
+#include "dali/kernels/alloc.h"
 #include "dali/pipeline/data/backend.h"
 #include <curand_kernel.h>  // NOLINT
 
 namespace dali {
 
-namespace detail {
+struct curand_states {
+  curand_states(uint64_t seed, size_t len);
+  ~curand_states() = default;
 
-template <typename T>
-struct CurandNormal {};
-
-template <>
-struct CurandNormal<float> {
-  __device__ static DALI_FORCEINLINE float normal(curandState *state) {
-    return curand_normal(state);
+  DALI_HOST_DEV inline curandState* states() {
+    return states_;
   }
-};
 
-template <>
-struct CurandNormal<double> {
-  __device__ static DALI_FORCEINLINE double normal(curandState *state) {
-    return curand_normal_double(state);
+  DALI_HOST_DEV inline curandState& operator[](size_t idx) {
+    assert(idx < len_);
+    return states_[idx];
   }
-};
 
-}  // namespace detail
+ private:
+  size_t len_;
+  int device_;
+  kernels::memory::KernelUniquePtr<curandState> states_mem_;
+  curandState* states_;  // std::unique_ptr::get can't be called from __device__ functions
+};
 
-class RandomizerGPU {
- public:
-  explicit RandomizerGPU(int seed, size_t len);
+template <typename T>
+struct curand_normal_dist {};
 
-  __device__ inline int rand(int idx) {
-    return curand(&states_[idx]);
-  }
+template <>
+struct curand_normal_dist<float> {
+  float mean = 0.0f, stddev = 1.0f;
 
-  template <typename T>
-  __device__ inline T normal(int idx) {
-    return detail::CurandNormal<T>::normal(&states_[idx]);
+  __device__ inline float yield(curandState *state) {
```

Comment:
Why `yield`? This name implies some kind of state - and this object doesn't have it. How about following C++ random distributions and having 
```suggestion
  __device__ inline float operator()(curandState *state) {
```
?

Reply:
will do